pyccel/codegen/printing/ccode.py

```diff
@@ -837,11 +837,31 @@ def _print_IndexedElement(self, expr):
                         inds[i] = Slice(ind, PyccelAdd(ind, LiteralInteger(1)), LiteralInteger(1))
                 inds = [self._print(i) for i in inds]
                 return "array_slicing(%s, %s, %s)" % (base_name, expr.rank, ", ".join(inds))
-            inds = [self._print(i) for i in inds]
+            inds = [self._cast_to(i, NativeInteger(), 8).format(self._print(i)) for i in inds]
         else:
             raise NotImplementedError(expr)
         return "%s.%s[get_index(%s, %s)]" % (base_name, dtype, base_name, ", ".join(inds))
 
+    def _cast_to(self, expr, dtype, precision):
+        """ add cast to an expresion when needed
+        parameters
+        ----------
+            expr      : PyccelAstNode
+                the expression to be casted
```

Comment:
I believe that past participle of "cast" is "cast". It is an irregular verb.

Reply:
> I believe

You are correct